File: Marcel/Classes/SoundMap.m
//  Copyright (c) 2014-present Ryan Gomba. All rights reserved.

#import "SoundMap.h"

@implementation SoundMap

+ (NSDictionary *)soundMap {
    static NSDictionary *map;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
        map = @{
                kButtonSounds: @[
                    [SoundSet soundSetWithSounds:@[@"uhh", @"ahh"]],
                    [SoundSet soundSetWithSounds:@[@"guess-bread", @"guess-pen", @"guess-raisin", @"guess-toenails"]],
                    [SoundSet soundSetWithSounds:@[@"hot-between", @"hot-zone"]],
                    [SoundSet soundSetWithSounds:@[@"treats-all", @"treats-look", @"treats-snoozin"]],
                    [SoundSet soundSetWithSounds:@[@"compared-to-what"]],
                    [SoundSet soundSetWithSounds:@[@"lentil", @"daily-life"]],
                    [SoundSet soundSetWithSounds:@[@"what-they-say"]],
                    [SoundSet soundSetWithSounds:@[@"grape-i-know", @"grape-bs", @"grape-but-you-could"]],
                    [SoundSet soundSetWithSounds:@[@"cheese", @"soda", @"lint", @"beg"]],
                    [SoundSet soundSetWithSounds:@[@"party"]],
                    [SoundSet soundSetWithSounds:@[@"alan"]],
                    [SoundSet soundSetWithSounds:@[@"muffin-a", @"muffin-sleep"]],
                ],
        };
    });
    return map;
}

@end

File: Classes/SoundMap.swift
import UIKit

class SoundMap: NSObject {
    public static let shared = SoundMap()
    public let soundSets: [SoundSet]

    override init() {
        self.soundSets = [
            SoundSet.init(sounds: ["uhh", "ahh"]),
            SoundSet.init(sounds: ["guess-bread", "guess-pen", "guess-raisin", "guess-toenails"]),
            SoundSet.init(sounds: ["hot-between", "hot-zone"]),
            SoundSet.init(sounds: ["treats-all", "treats-look", "treats-snoozin"]),
            SoundSet.init(sounds: ["compared-to-what"]),
            SoundSet.init(sounds: ["lentil", "daily-life"]),
            SoundSet.init(sounds: ["what-they-say"]),
            SoundSet.init(sounds: ["grape-i-know", "grape-bs", "grape-but-you-could"]),
            SoundSet.init(sounds: ["cheese", "soda", "lint", "beg"]),
            SoundSet.init(sounds: ["party"]),
            SoundSet.init(sounds: ["alan"]),
            SoundSet.init(sounds: ["muffin-a", "muffin-sleep"]),
        ]
        super.init()
    }

}
